lib/auth/token/jwt.dart

```diff
@@ -17,11 +21,30 @@ class Jwt {
 
   String get userId => _payload.subject;
 
-  Jwt(String token) : _tokenParts = token.split(' ').last.split('.');
+  Jwt(this.token) : _tokenParts = token.split(' ').last.split('.');
+
+  Future<bool> validate(String projectId, Map<String, String> certificates,
+      {bool enforceEmailVerification = false,
+      bool checkRevoked = false}) async {
+    // Google can validate the token for us and we can compare the results to revoke it.
+    // Otherwise, do local validation so we don't use network bandwidth and deal with latency.
+    if (checkRevoked) {
+      var user = await FirebaseAuth.instance.getUser(uid: userId);
+
+      if (user.tokensValidAfterTime != null) {
+        var issuer = await Issuer.discover(Issuer.firebase(projectId));
+        var client = Client(issuer, projectId);
+
+        var credential = client.createCredential(idToken: token);
+
+        final authTimeUtc = credential.idToken.claims.authTime;
+        final validSinceUtc = user.tokensValidAfterTime;
+        if (authTimeUtc.isBefore(validSinceUtc)) {
+          throw RevokedTokenException();
+        }
+      }
```

Comment:
This code is useless. If the token is invalid, the `getUser` call above will fail.

Reply:
Yes, which is good. that means verifyToken will fail.

However, if the getUser call does not fail, we want to make sure the token is not revoked.

Reply:
If you make the getUser request with the token being tested, it will fail if it's revoked.

Reply:
That's the point though! Isn't it? My point to moving this here is that, since google can check if it's valid or not, we might as well let them handle the whole thing and relieve some computation on our end by not doing local verification.

If the user wants to check if it's revoked, let google check the validity of the whole thing as well, we don't need to verify it locally, and THEN let google verify it AGAIN. That's a waste of resources. Let google do the whole thing.

Otherwise, if we don't want to check if it's revoked, check locally.

Reply:
That was the point I made a few hours ago.
I still don't understand everything that comes after getUser

Reply:
Google validates the token and checks the user, but doesn't throw an exception if it's a revoked token.
getUser throwing an exception will mean the token is invalid. 
Then we have to check if it's revoked or not manually.

the java admin sdk also does this.
![image](https://user-images.githubusercontent.com/7407478/82753253-f525b600-9dcc-11ea-83ea-8ffc4c3604e5.png)

![image](https://user-images.githubusercontent.com/7407478/82753256-f9ea6a00-9dcc-11ea-807d-7dadcdecb1ef.png)

Reply:
> Google validates the token and checks the user, but doesn't throw an exception if it's a revoked token.

Have you tested this?

Reply:
I planned on testing everything after you finished reviewing it all.

Reply:
A PR should be a tested, finished piece of code. What's the point of reviewing code that might not work?

Reply:
Do you really want me to answer that question, given the amount of discussion we had to have here... Hahaha

I'll mark this PR as a draft. That's the proper usage of that feature.